lib/elixir/lib/enum.ex

```diff
@@ -2855,7 +2857,7 @@ defimpl Enumerable, for: Map do
     {:ok, false}
   end
 
-  def reduce(map, acc, fun) do
+  def reduce(map, acc, fun) when is_function(fun, 2) do
```

Comment:
Had doubts about this one, as I believe I don't fully understand in which cases this function is called. Removed the guard check.